### ISSUE TITLE
Native source packages are not supported

### ISSUE THREAD

**Issue by Shnatsel** [rcrowley/freight#47]
Source packages native to the distribution, i.e. consisting of just .dsc and .tar.gz instead of .dsc+.orig.tar.gz+.debian.tar.gz, are not supported by Freight. It seems to recognize only the non-native packages, as per https://github.com/rcrowley/freight/blob/master/bin/freight-add#L32

Adding such .dsc files are treated as "invalid" and attempting to add .tar.gz file prints usage.

This is a serious limitation because distribution-native packages are quite common.

**Comment by rcrowley** [comment]
@Shnatsel can you test this out, please?

**Comment by Shnatsel** [comment]
It doesn't work, fails on trying to mkdir over the .tar.gz

Here's the debug log: http://pastebin.com/8ENspB1U

**Comment by rcrowley** [comment]
That debug log is an incorrect invocation of `freight-add`, probably.  The usage message says it takes a single package (which, for source packages constitutes two or three arguments) and a list of archives in which that package should be included.  For example:

```
freight add foo_0.0.0-1.deb apt/precise
freight add bar_0.0.0-1.dsc bar_0.0.0-1.tar.gz apt/precise
```

Does running it like that make any difference?  It looks like this patch works for me but I want to make sure it works for you, too.

**Comment by Shnatsel** [comment]
I've run it like this:
`freight add build_folder/* apt/precise`
The folder contained .deb, .dsc and .tar.gz as produced by a single pbuilder run. I guess this is the problem. I'm sorry I can't test this invocation now - I'm leaving for FOSDEM in two hours. 

If this is not supported it should be at least documented - oh, I've already complained about that in another issue... 

Other than that, thanks for fixing it so quickly! I'll update my build scripts to use this as soon as it makes its way into the Debian repository.

**Comment by rcrowley** [comment]
I'll get that documented and released soon, thanks for confirming.  I'm not likely to change the interface to accept multiple packages in one call because I think the more common (and if not incompatible at least confusing) invocation is to add the same package to multiple archives.

**Comment by rcrowley** [comment]
Released in v0.3.5.
